pkg/appmgmt/general/general_test.go

```diff
@@ -526,3 +527,269 @@ func TestGetOwnerReferences(t *testing.T) {
 	assert.Equal(t, returnedOwnerRefs[0].Kind, "Pod", "Unexpected owner reference Kind")
 	assert.Equal(t, returnedOwnerRefs[0].APIVersion, v1.SchemeGroupVersion.String(), "Unexpected owner reference Kind")
 }
+
+// nolint: funlen
+func TestListApplication(t *testing.T) {
+	var app01, app02, app03, app04, app05, app06 = "app00001",
+		"app00002", "app00003", "app00004", "app00005", "app00006"
+	var queue01, queue02 = "root.queue01", "root.queue02"
+	var ns01, ns02 = "namespace01", "namespace02"
+
+	// mock the pod lister for this test
+	mockedAPIProvider := client.NewMockedAPIProvider()
+	mockedPodLister := test.NewPodListerMock()
+	mockedAPIProvider.SetPodLister(mockedPodLister)
+
+	// app01 pods, running in namespace01 and queue01
+	// all pods are having applicationID and queue name specified
+	// 2 pods have assigned nodes, 1 pod is pending for scheduling
+	// allocated pod
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app01pod00001",
+			Namespace: ns01,
+			Labels: map[string]string{
+				constants.LabelApplicationID: app01,
+				constants.LabelQueueName:     queue01,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+			NodeName:      "allocated-node",
+		},
+	})
+
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app01pod00002",
+			Namespace: ns01,
+			Labels: map[string]string{
+				constants.LabelApplicationID: app01,
+				constants.LabelQueueName:     queue01,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+			NodeName:      "allocated-node",
+		},
+	})
+
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app01pod00003",
+			Namespace: ns01,
+			Labels: map[string]string{
+				constants.LabelApplicationID: app01,
+				constants.LabelQueueName:     queue01,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+		},
+	})
+
+	// app02 pods, running in queue02 and namespace02
+	// 2 pods are having applicationID and queue name specified
+	// both 2 pods are pending
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app02pod0001",
+			Namespace: ns02,
+			Labels: map[string]string{
+				constants.LabelApplicationID: app02,
+				constants.LabelQueueName:     queue02,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+		},
+	})
```

Comment:
This is a duplicate from app01pod00003 and does not test anything new, should be removed.